lib/iris/fileformats/grib/_message.py

```diff
@@ -89,38 +93,42 @@ def data(self):
 
         """
         sections = self.sections
-        grid_section = sections[3]
-        if grid_section['sourceOfGridDefinition'] != 0:
-            raise TranslationError(
-                'Unsupported source of grid definition: {}'.format(
-                    grid_section['sourceOfGridDefinition']))
-
-        reduced = (grid_section['numberOfOctectsForNumberOfPoints'] != 0 or
-                   grid_section['interpretationOfNumberOfPoints'] != 0)
-        template = grid_section['gridDefinitionTemplateNumber']
-        if reduced and template not in (40,):
-            raise TranslationError('Grid definition Section 3 contains '
-                                   'unsupported quasi-regular grid.')
-
-        if template in (0, 1, 5, 12, 30, 40, 90):
-            # We can ignore the first two bits (i-neg, j-pos) because
-            # that is already captured in the coordinate values.
-            if grid_section['scanningMode'] & 0x3f:
-                msg = 'Unsupported scanning mode: {}'.format(
-                    grid_section['scanningMode'])
-                raise TranslationError(msg)
-            if template in (30, 90):
-                shape = (grid_section['Ny'], grid_section['Nx'])
-            elif template == 40 and reduced:
-                shape = (grid_section['numberOfDataPoints'],)
+        if self.edition == 1:
+            grid_section = sections[2]
+            shape = (grid_section['Nj'], grid_section['Ni'])
```

Comment:
So the key question for this implementation, IMO, is - is this in the spirit of coded keys only.

I am, of course, asking as a complete novice to GRIB.